### DISCUSSION TITLE
Removing pores above or below a size threshold

### DISCUSSION THREAD

**Discussion by baq99** [PMEAL/OpenPNM#2023] [Q&A] [answered]
Hello everyone,

Does anyone know how I would go about removing all pores above or below a volume or diameter threshold so I can output a pore size distribution, total pore volume or geometry minus those values?

I have extracted a SNOW network from tomography using PoreSpy but want to edit out pores above and below a specific size to calculate pore volumes within a range.  I imagine there's an easier way than editing the geometry, maybe just filtering the output?

Also can we only plot pore size distribution histograms?  Are there options for line graphs instead?

**Comment by ma-sadeghi** [comment, marked as answer]
Hi, it's actually quite straightforward:

```python
mask = net["pore.diameter"] > threshold
op.topotools.trim(network, pores=mask)
```

Note: it's crucial that you do the query on your network object, rather than your geometry, as network indices are global whereas geometry indices are local (of course, if you only have only geometry, then the indices are identical, but for more than 1 geometry, you get different results).

**Reply by baq99** [reply]
I'll give this a try.  Thank you so much for getting back to me.

**Reply by baq99** [reply]
It worked, thank you very much for your reply.

**Comment by mourendong** [comment]
Hi, @ma-sadeghi, the way that removes pores or throats that will change the serial number of pores or throats. 
I would go about labeling the pores or throats clusters, and the isolated pores or throats which were not connected with others were label as cluster1, cluster2, ... and so on. 
I plan to use the way mentioned above, first, I labeled the connected pores, and then remove these. And using the 'check_network_health()' to find all disconnected_clusters. But, as I remove the pores or throats of the network, the serial number of pores or throats was changed, which lead to I can not compare with the original. 
Can you give me some advice on how to save this question or how to label these clusters?

**Reply by jgostick** [reply]
From what I gather based on your intial comment, you didn't want to delete the pores, you just wanted to the the size distribution of pores below a certain size.  You can use @ma-sadeghi's suggestion of find a ``mask``, then using matplotlib to show the sizes, like ``plt.hist(pn['[pore.volume'][mask])``.

Regarding adding a fixed serial number, you can always assigned a number *before* deleting, then the surviving pores will still have their number, like ``pn['pore.serial'] = np.range(pn.Np)``, then you can still find pores with x, y, or z as a serial number assuming they haven't been trimmed.

**Reply by mourendong** [reply]
I'll give this a try. Thanks for your kind reply!
